components/plurals/src/lib.rs

```diff
@@ -359,4 +339,17 @@ impl PluralRules {
     pub fn select<I: Into<PluralOperands>>(&self, input: I) -> PluralCategory {
         self.selector.select(&input.into())
     }
+
+    /// Lower-level constructor that allows constructing a PluralRules directly from
+    /// data obtained from a provider
+    pub fn new_from_data<'d>(
+        langid: LanguageIdentifier,
+        data: &PluralRuleStringsV1<'d>,
```

Comment:
I think this should be taken by value.  I think in most cases we should take ownership of the data struct in order to avoid unnecessary clones.  It's not useful for the caller to keep the data struct around any longer than when they call `new_from_data`.

Reply:
There's no unnecessary clone happening here, is there? We're parsing the data struct.

Reply:
If the data struct owns its own strings, then passing by `&` means that they need to be cloned in the general case.  For PluralRuleStringsV1 in particular, maybe this doesn't matter.

I guess the question is whether we should make these methods "always take by value", "always take by reference", or "case-by-case basis"?

Reply:
Bear in mind, we usually get a Cow out, so if we wanted to take ownership we would be _forced_ to clone

Reply:
Filed https://github.com/unicode-org/icu4x/issues/615, once that is resolved we can revisit this since the data model will change.